src/lib/site-server-process-child.ts

```diff
@@ -0,0 +1,72 @@
+import { PHPRunOptions } from '@php-wasm/universal';
+import { startServer, type WPNowServer } from '../../vendor/wp-now/src';
+import { WPNowOptions } from '../../vendor/wp-now/src/config';
+import { setupLogging } from '../logging';
+import type { MessageName } from './site-server-process';
+
+// Setup logging for the forked process
+if ( process.env.STUDIO_APP_LOGS_PATH ) {
+	setupLogging( {
+		processId: 'site-server-process',
+		isForkedProcess: true,
+		logDir: process.env.STUDIO_APP_LOGS_PATH,
+	} );
+}
+
+const options = JSON.parse( process.argv[ 2 ] ) as WPNowOptions;
+let server: WPNowServer;
+
+async function start( message: string, messageId: number ) {
+	server = await startServer( options );
+	process.parentPort.postMessage( {
+		message,
+		messageId,
+		data: {
+			php: {
+				documentRoot: server.php.documentRoot,
+			},
+		},
+	} );
+}
+
+async function stop( message: string, messageId: number ) {
+	await server.stopServer();
+	process.parentPort.postMessage( {
+		message,
+		messageId,
+	} );
+}
+
+async function runPhp( message: string, messageId: number, data: PHPRunOptions ) {
+	const response = await server.php.run( data );
+	process.parentPort.postMessage( { message, messageId, data: response.text } );
+}
+
+process.parentPort.on( 'message', async ( { data: messagePayload } ) => {
+	const { message, messageId, data }: { message: MessageName; messageId: number; data: unknown } =
+		messagePayload;
+	try {
+		switch ( message ) {
```

Comment:
[Suggestion]: 
1. Extract messages in order to avoid magic strings; eg: 
```
const messages = { START_SERVER: 'start-server' ... 
```
or
2. Extract messages along with their associated functions; Eg:
```
const messageHandlers = {
    'start-server': start,
    'stop-server': stop,
    'run-php': runPhp
};
....
    try {
        const handler = messageHandlers[message];
        if (!handler) {
             return
        }

        await handler(message, messageId, data);
    } catch (error) {
        const errorObj = error instanceof Error ? error : new Error('Unknown Error');
        postMessage(errorObj);
    }
```
We can further decouple by extracting the creating a `emitMessage` function like so:

```
    const emitMessage = (error) => {
        const response = { message, messageId };
        if (error) {
            response['error'] = error;
        }
        process.parentPort.postMessage(response);
    };
```
and remove it from the handlers like so:
```
async function runPhp( message: string, messageId: number, data: PHPRunOptions ) {
	await server.php.run( data );
}
```
and use it like so:

```
   const emitMessage = (error = null) => {
        const response = { message, messageId };
        if (error) {
            response['error'] = error;
        }
        process.parentPort.postMessage(response);
    };

    try {
        const handler = messageHandlers[message];
        if (!handler) {
            return
        }

        await handler(message, messageId, data);
        emitMessage();
    } catch (error) {
    	const errorObj = error as Error;
		if ( ! errorObj ) {
			emitMessage( Error( 'Unknown Error' ) );
			return;
		}
		emitMessage(  errorObj);
    }
```

Reply:
@kozer following your suggestion, I updated this file in https://github.com/Automattic/studio/pull/19/commits/35666628a412e8ffcf2d1d62d5f20b6c8c0a685e. Let me know if it's aligned with the changes you proposed, thanks 🙇 !